### ISSUE TITLE
DeviceTokenFormatValidator preventing notification save on iOS Apnsp8 app using Firebase Token

### ISSUE THREAD

**Issue by ebkcode** [rpush/rpush#590]
**Describe the bug**
When trying to create/save a notification with an iOS Apnsp8 app, (Rpush::Apnsp8::App) the save fails due to a validation on the device_token from validate method in DeviceTokenFormatValidator. (The token provided is automatically generated from the Firebase SDK in an installed app on the device.)

**To Reproduce**
1) Write code to create/save a notification with an iOS Apnsp8 app, using a Firebase token.
```
n = Rpush::Apnsp8::Notification.new
n.app = Rpush::Apnsp8::App.find_by_name('ios_app')
n.device_token = "<token from Firebase here, see example below>"
# Example Firebase Token: dLixI0g6qkkYk5spuu3zMt:APA92bEWyh9fuYhChItPT0UvQSzwNHIWVpErSXQqxudHpE55BfK_6dERg556jNT7KWqRNh6BZrTxpZR5mNydj1bs95O2u0w3Pkb4dwqp1f6QkJkq7AmJuHgvr-0d62-ymlygBhaemv6K
n.alert = "Test Push Notification"
n.save!
```
2) Execute code
3) See error: ActiveRecord::RecordInvalid ... Validation failed: Device token is invalid

**Expected behavior**
Notification should be saved and sent.

**System configuration (please complete the following information):**
 - OS: macOS Big Sur
 - OS version: 11.2.1
 - Ruby version: 2.4.1
 - Rails version: 5.1.7
 - Rpush version: 5.3.0

**Additional context**
Using Firebase Cloud Messaging (FCM)

**Comment by ebkcode** [comment]
Need APNSToken (hex string) instead of "Firebase Token"

**Comment by Sohair63** [comment]
@ebkcode can you confirm if this is what we can convert using firebase token?

**Comment by ebkcode** [comment]
@Sohair63 - I'm not sure exactly what you're asking, but I can confirm that to use the Apnsp8 Rpush API, one must pass in the APNSToken (hex string) from their iOS app, rather than the Firebase Token which is used for the FCM/GCM Android Rpush implementation.

The example documentation in the README does note this with a comment, as shown, but it's not clearly noted which "token" to use: `n.device_token = "..." # hex string` As a result, once I got it working, I closed the issue. However, I'm hopeful that this post might help others if they are having trouble. 

To get the device_token needed for Rpush, one can convert the firebase token like so:
`NSString *apnsTokenString = APNSStringForAPNSDeviceToken([FIRMessaging messaging].APNSToken);`

```
NSString *APNSStringForAPNSDeviceToken(NSData *deviceToken) {
  NSMutableString *APNSToken = [NSMutableString string];
  unsigned char *bytes = (unsigned char *)[deviceToken bytes];
  for (int i = 0; i < (int)deviceToken.length; i++) {
    [APNSToken appendFormat:@"%02x", bytes[i]];
  }
  return APNSToken;
}
```
